app/page.jsx

```diff
@@ -1,33 +1,70 @@
 "use client";
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 import Header from './components/header';
 import Form from './components/form';
 import TaskList from './components/TaskList';
 
 const Page = () => {
-  const [mainTask, setMainTask] = useState([]);
+  const [fetchedTasks, setFetchedTasks] = useState([]);  
+  const [localTasks, setLocalTasks] = useState([]);    
+
+
+  useEffect(() => {
+    const fetchTasks = async () => {
+      try {
```

Comment:
It's good that you used try and catch.

But we have to write it in a separate file. we'll see more soon.